corva/models/stream.py

```diff
@@ -1,27 +1,75 @@
-from typing import Any, Dict, List, Optional
+from __future__ import annotations
 
-from corva.models.base import BaseContext, BaseEventData
+from typing import Dict, List, Optional
+
+from pydantic import parse_raw_as
+
+from corva.models.base import BaseContext, BaseEventData, ListEvent
 from corva.state.redis_state import RedisState
 
 
 class StreamContext(BaseContext):
     state: RedisState
 
 
+class RecordData(BaseEventData):
```

Comment:
I don't think we can do this. `data` inside of a Record is variable, depending on the dataset.

Reply:
I've accounted for missed, forgotten or added in future fields by setting `BaseEventData.Config.extra = allow` [link](https://github.com/corva-ai/python-sdk/blob/a715bf4d4cffe06f1794c20fc6e6ddc83c96a8ae/corva/models/base.py#L37)